src/requests/consumption.js

```diff
@@ -0,0 +1,377 @@
+		try {
+			console.log(templateData, 'templateData')
+			// Format the template
+			let formattedTemplate = formatTemplate(templateData)
+			if (!formattedTemplate.success) {
+				throw new Error('FAILED_TO_FORMAT_TEMPLATE')
+			}
+
+			let template = formattedTemplate.template
+
+			// Process Categories
+			if (templateData.categories?.length > 0) {
+				let categoriesResponse = await processCategories(templateData.categories)
+				if (!categoriesResponse.success) {
+					throw new Error('FAILED_TO_FETCH_OR_CREATE_CATEGORIES')
+				}
+				template.categories = categoriesResponse.categories
+			}
+
+			// Insert the template into the database
+			const templateCollection = mongoDb.collection(COLLECTIONS.TEMPLATES)
+			const result = await templateCollection.insertOne(template)
+
+			// Validate the result of the template creation
+			if (!result || !result.insertedId) {
+				throw new Error('Failed to insert the template into the database.')
+			}
+
+			const templateId = result.insertedId
+
+			// Process and Create Tasks
+			const processedTasks = assignSequenceNumbers(templateData.tasks || [])
+			const taskCreationResponse = await createTasks(processedTasks, templateId, template.externalId)
+
+			// Validate the result of the task creation
+			if (!taskCreationResponse.success) {
+				throw new Error('FAILED_TO_CREATE_TASKS')
+			}
+
+			// Update Template with tasks and sequence
+			await templateCollection.updateOne(
+				{ _id: templateId },
+				{
+					$set: {
+						tasks: taskCreationResponse.taskIds,
+						taskSequence: taskCreationResponse.externalIds,
+					},
+				}
+			)
+
+			//update the published id in resource table
+			await resourceService.publishCallback(templateData.id, templateId.toString())
+
+			//return result
+			result.success = true
+			result.templateId = templateId
+			return resolve(result)
+		} catch (error) {
+			result.error = `Error: ${error.message}`
+			return reject(error)
+		}
+	})
+}
+
+/**
+ * Format Project Template
+ * @name formatTemplate
+ * @param {Object} templateData - Project template data
+ * @returns {Object} - Response contains formatted template
+ */
+const formatTemplate = (templateData) => {
+	try {
+		let template = {
+			title: templateData.title,
+			description: templateData.objective || '',
+			keywords: templateData.keywords ? templateData.keywords.split(',').map((k) => k.trim()) : [],
+			isDeleted: false,
+			recommendedFor: templateData.recommended_for?.length
+				? templateData.recommended_for.map((item) => item.label)
+				: [],
+			createdBy: templateData.user_id,
+			updatedBy: templateData.user_id,
+			learningResources: templateData.learning_resources ? convertResources(templateData.learning_resources) : [],
+			isReusable: true,
+			taskSequence: [], // Initially empty
+			deleted: false,
+			status: common.PUBLISHED_STATUS,
+			externalId: generateExternalId(templateData.title),
+			entityType: '',
+			metaInformation: {
```

Comment:
Can we add goal , rationale etc... from the template data? Just in case in future we have those fields coming from SCP we need not make change here.  
`
templateDetails?.goal ? templateDetails?.goal  : ''
`

@priyanka-TL

Reply:
@adithyadinesh0412 If the key is different in scp, again we need to make a code change, i think we can update this later